Add DD_Version to php for deployment tracking

ahosking opened the pull request:
This seems to consistently enable us to include the hash of the running JBX code to send to Datadog for better error tracking and APM.
<img width="544" alt="Screen Shot 2022-06-11 at 12 57 23 AM" src="https://user-images.githubusercontent.com/1209293/173173464-b3cf1856-c860-4d32-950a-725a210bab83.png">

Commit:
Add DD_Version to php for deployment tracking

ahosking commented:
I should note that it may also be reasonable to use this as a provider for the deployment markers recipe as well.

Bellardia commented on configure/recipes/default.rb:
```diff
@@ -1,3 +1,14 @@
+node.default[:commit_hash] = "null"
+
+ruby_block "jbx_version" do
+  block do
+    %s(su #{node[:user]; git config --global --add safe.directory #{node["jbx"]["path"]})
+    commit_hash = %x(cd #{node["jbx"]["path"]}; git rev-parse HEAD)
+    node.default[:commit_hash] = "#{commit_hash.strip}"
+  end
+  action :run
+end
```
can you localize this to the jbx cookbook, we shouldn't assume `jbx` is configured in all paths

Bellardia commented on configure/recipes/default.rb:
```diff
@@ -1,3 +1,14 @@
+  block do
+    %s(su #{node[:user]; git config --global --add safe.directory #{node["jbx"]["path"]})
+    commit_hash = %x(cd #{node["jbx"]["path"]}; git rev-parse HEAD)
+    node.default[:commit_hash] = "#{commit_hash.strip}"
```
You want to use `run_state` here rather than `node.default`
https://docs.chef.io/recipes/#noderun_state

```
node.run_state[:jbx_version] =
```

ahosking replied:
Ah, ok. I'll go back and switch it out.
I spent a whack of time on options that may have worked but, I originally had the block in the wrong location where it wasn't actually passing the value onto further configurations.
I'll make this change.

Bellardia commented on configure/templates/default/www.conf.erb:
```diff
@@ -380,6 +380,7 @@ env[DD_PROFILING_ENABLED] = <%= (node[:datadog][:enable_profiling] || 0) && "tru
 env[DD_TRACE_ENABLED] = <%= (node[:datadog][:enable_trace_agent] || 0) && "true" || "false" %>
 env[DD_AGENT_HOST] = $HOSTNAME
 env[DD_SERVICE] = <%= node.read("datadog", "tags", "service") || "dev" %>
+env[DD_VERSION] = <%= (node["commit_hash"]) || "env[DD_VERSION] = lazy {node[commit_hash]}" %>
```
this code is a bit suspect
You want to use `lazy` inside cookbooks/configure/recipes/php.rb instead:
```
variables(
    lazy {
      node["php"]["fpm"].merge({
         commit_hash: node.run_state[:jbx_version]
      })
   }
)
```

then in the template you can do 
```
<%= @commit_hash %>
```

ahosking replied:
Well, see, I didn't know I could do this with a `merge`, that's a pro-tip! 👏 

> ```
> variables(
>     lazy {
>       node["php"]["fpm"].merge({
>          commit_hash: node.run_state[:jbx_version]
>       })
>    }
> )
> ```

Commit:
Change variables to run_state and contextualize to jbx

Make template easier to read with named variable
Move variable definition to JBX context

ahosking reviewed (commented):
@Bellardia I've made changes based on your suggestions and have tested for success!

ahosking commented:
@Bellardia, I've moved this to a draft as it doesn't work in opsworks.

Commit:
Update www.conf.erb

Add default value when commit hash doesn't get passed in

Commit:
Update template to include version only when not empty

Commit:
Pull out the fpm config changes

Commit:
Move the Datadog Version to nginx

ahosking commented:
@Bellardia Your ini tweak works great, good call!
<img width="841" alt="image" src="https://user-images.githubusercontent.com/1209293/175966620-6f159a54-60f6-4f79-9b5c-f813eb1d0a49.png">

Bellardia commented on jbx/templates/default/api.erb:
```diff
@@ -79,6 +79,7 @@ server {
         fastcgi_param PATH_INFO         $fastcgi_path_info;
         fastcgi_param PATH_TRANSLATED   $document_root$fastcgi_path_info;
         fastcgi_param SCRIPT_FILENAME   $document_root$fastcgi_script_name;
+        fastcgi_param DD_VERSION	<%= "#{node.run_state[:jbx_version]}" %>;
```
```suggestion
        fastcgi_param DD_VERSION	<%= node.read("run_state", "jbx_version") || node[:environment] %>;
```

Commit:
Update jbx/templates/default/api.erb

Co-authored-by: Bellardia <ian@jumbleberry.com>
